src/utils/helper.ts

```diff
@@ -80,6 +97,18 @@ export function formatDecimalUnits(value: BigDecimal, decimals: BigInt): BigDeci
 	return value.div(scaler.toBigDecimal());
 }
 
+
+export function calcWeightedAverage(values: BigInt[], weights: BigInt[]): BigInt {
+
+	let numerator = BigInt.fromI32(0);
+	let denominator = BigInt.fromI32(0);
+	for (let i = 0; i < values.length; i++) {
+		denominator = denominator.plus(values[i].times(weights[i]))
```

Comment:
im gonna write a local script to test this theory out but I feel like we need to normalise the weighting here or else end up with a bigger number than we want since it will be multiple massive integers